spec/requests/rooms_request_spec.rb

```diff
@@ -9,13 +9,33 @@
     post '/validate_beta_user', params: { beta_code: 'correct-code' }
   end
 
-  let(:room) { create(:room) }
   let(:current_jam) { create(:jam, room: room) }
+  let(:room) { build(:room) }
+  let(:room_params) { { room: { name: room.name } } }
 
+  context "user is authenticated" do
+    before { sign_in create(:user) }
+    let(:room) { build(:room, name: nil) }
+
+    context 'invalid room' do
+      let(:room) { build(:room, name: '') }
+
+      it 'redirects to the previous page user was on' do
```

Comment:
ah, you do it here, but it really belongs to the controller specs since it's the controller level not view